docs/csharp/csharp-6.md

```diff
@@ -1,17 +1,570 @@
+If the expression used for an exception filter evaluates to `true`, the
+catch clause performs its normal procesing on an exception. If the
+expression evaluates to `false`, then the `catch` clause is skipped.
+
+One use is to examine information about an exception to determine if a
+catch clause can process the exception:
+
+[!code-csharp[ExceptionFilter](../../samples/snippets/csharp/new-in-6/NetworkClient.cs#ExceptionFilter)]
+
+The code generated by exception filters provides better information about
+an exception that is thrown and not processed. Before exception filters
+were added to the language you would need to create code like the following:
+
+[!code-csharp[ExceptionFilterOld](../../samples/snippets/csharp/new-in-6/NetworkClient.cs#ExceptionFilterOld)]
+
+The point where the exception is thrown changes between these two examples.
+In the previous code, where a `throw` clause is used, any stack trace
+analysis or examination of crash dumps will show that the exception was
+thrown from the `throw` statement in your catch clause. The actual exception
+object will contain the original call stack, but all other information
+about any variables in the call stack between this throw point and the
+location of the original throw point has been lost. 
+
+Contrast that with how the code using an exception filter is processed:
+The exception filter expression evaluates to `false`. Execution never
+enters the `catch` clause. No stack unwinding takes place. The original
+throw location is preserved for any debugging activities that would take
+place later.
+
+Whenever you need to evaluate fields or properties of an exception, instead
+of relying on solely on the exception type, use an exception filter to
+preserve more debugging information.
+
+Another recommended pattern with exception filters is to use them for
+logging routines. This usage also leverages the manner in which the exception
+throw point is preserved when an exception filter evaluates to `false`.
+
+A logging method would be a method whose argument is the exception that
+unconditionally returns `false`:
+
+[!code-csharp[ExceptionFilterLogging](../../samples/snippets/csharp/new-in-6/ExceptionFilterHelpers.cs#ExceptionFilterLogging)]
+
+Whenever you want to log an exception, you can add a catch clause, and
+use this method as the exception filter:
+
+[!code-csharp[LogException](../../samples/snippets/csharp/new-in-6/program.cs#LogException)]
+
+The exceptions are never caught, because the `LogException` method always
+returns `false`. That always false exception filte means that you can
+place this logging handler before any other exception handlers:
+
+[!code-csharp[LogExceptionRecovery](../../samples/snippets/csharp/new-in-6/program.cs#LogExceptionRecovery)]
+
+The above example highlights a very important facet of exception filters.
+The exception filters enable scenarios where a more general exception
+catch clause may appear before a more specific one. It's also possible
```

Comment:
Yes. That's mentioned in issue #274

Is there another topic that should be updated as well?